types/index.d.ts

```diff
@@ -0,0 +1,28 @@
+import Vue = require('vue')
+import { WatchOptions } from 'vue'
+import { Observable } from 'rxjs/Observable'
+
+export type Observables = Record<string, Observable<any>>
+declare module 'vue/types/options' {
+  interface ComponentOptions<V extends Vue> {
+    subscriptions?: Observables | ((this: V) => Observables)
+    domStreams?: string[]
+  }
+}
+
+declare module "vue/types/vue" {
+  interface Vue {
+    $observables: Observables;
+    $watchAsObservable(exprOrFn: string | Function, options?: WatchOptions): Observable<any>
```

Comment:
I think we can improve this type by using type parameter like `$watch`.

```ts
$watchAsObservable<T>(fn: (this: this) => T, options?: WatchOptions): Observable<T>
```

Reply:
Maybe, returned Observable has a value that is formed of `{ newValue: T, oldValue: T }`?

https://github.com/vuejs/vue-rx#watchasobservableexporfn-options